api/envoy/config/filter/http/header_to_metadata/v2/header_to_metadata.proto

```diff
@@ -20,6 +20,11 @@ message Config {
   enum ValueType {
     STRING = 0;
     NUMBER = 1;
+
+    // `Base64Url <https://tools.ietf.org/html/rfc4648#section-5>`_ encoded string of
+    // serialized `protobuf.Value
+    // <https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/struct.proto#L62>`_
+    BASE64URL_SERIALIZED = 2;
```

Comment:
This is counter intuitive from API stand point without reading the comment above, from reading the implementation, I'd say make this `VALUE` or `PROTOBUF_VALUE`, and have another enum for the encoding to express it, since which base64 or not is orthogonal to value type here, WDYT?

Reply:
That looks a cleaner API though I imagine the `base64` option is likely to be always used only with `PROTOBUF_VALUE`?

 I'm fine with the suggestion, @rgs1 WDYT?

Reply:
It is possible for someone to use STRING with BASE64 with some non header safe character sets (UTF-8), and when you add more binary types here it will help anyway.

Reply:
yeah i like Lizan's suggestion (`PROTOBUF_VALUE`).

Reply:
Thanks, updated.